mlflow/utils/search_utils.py

```diff
@@ -89,8 +89,14 @@ class SearchUtils:
     }
 
     @classmethod
-    def get_sql_filter_ops(cls, column, operator):
-        sql_filter_ops = {"LIKE": column.like, "ILIKE": column.ilike}
+    def get_sql_filter_ops(cls, column, operator, dialect):
+        if dialect == "mysql":
+            like_op = column.op("LIKE BINARY")
+        elif dialect == "mssql":
+            like_op = column.collate("SQL_Latin1_General_CP1_CS_AS").like
```

Comment:
Do we want to include Kana sensitive as well? 
"SQL_Latin1_General_CP1_CS_AS_KS" or does sorting based on characters make the ordering confusing?

Reply:
I think we do :)

Reply:
Updated!

Reply:
`SQL_Latin1_General_CP1_CS_AS_KS` didn't work:

https://github.com/mlflow/mlflow/runs/5728726867?check_suite_focus=true#step:11:10259

```
E           mlflow.exceptions.MlflowException: (pyodbc.ProgrammingError) ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid collation 'SQL_Latin1_General_CP1_CS_AS_KS'. (448) (SQLExecDirectW)")
```

Reply:
Is `_KS` only available for Japanese collations such as `Japanese_Bushu_Kakusu_100_CS_AS_KS_WS_UTF8`?

Reply:
I don't think kana-insensitivity is a big issue though.

Reply:
Offline discussion: we should also support kana sensitivity if there is no downside.

Reply:
I'm wondering if the `_UTF8` addition as suffix is required. It's pretty hard to tell from their docs... https://docs.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver15#Collation_Defn
I wonder if people using MSSQL as a tracking backend end up writing a T-SQL script when they setup MLflow to enable faster filtering and apply specific column collations to speed it up?

I agree, not that big of a deal to enable this (it would just be nice to enable functionality for as many users as possible, but if the config is broken, then it's broken...)